source/speech/manager.py

```diff
@@ -309,26 +309,34 @@ def _processSpeechSequence(self, inSeq: SpeechSequence):
 
 		def ensureEndUtterance(seq: SpeechSequence):
 			# We split at EndUtteranceCommands so the ends of utterances are easily found.
+			# This function ensures the given sequence ends with an EndUtterance command,
+			# Ensures that the sequence also includes an index command at the end,
+			# It places the complete sequence in outSeqs,
+			# It clears the given sequence list ready to build a new one,
+			# And clears the paramsToReplay list
+			# and refills it with any params that need to be repeated if a new sequence is going to be built.
 			if seq:
 				# There have been commands since the last split.
-				outSeqs.append(seq)
-				lastOutSeq = seq
+				lastOutSeq = paramsToReplay + seq
+				outSeqs.append(lastOutSeq)
+				paramsToReplay.clear()
+				seq.clear()
 				# Re-apply parameters that have been changed from their defaults.
-				seq = paramTracker.getChanged()
+				paramsToReplay.extend(paramTracker.getChanged())
 			else:
 				lastOutSeq = outSeqs[-1] if outSeqs else None
 			lastCommand = lastOutSeq[-1] if lastOutSeq else None
 			if not lastCommand or isinstance(lastCommand, (EndUtteranceCommand, ConfigProfileTriggerCommand)):
 				# It doesn't make sense to start with or repeat EndUtteranceCommands.
 				# We also don't want an EndUtteranceCommand immediately after a ConfigProfileTriggerCommand.
-				return seq
+				return
 			if not isinstance(lastCommand, IndexCommand):
 				# Add an index so we know when we've reached the end of this utterance.
 				reachedIndex = next(self._indexCounter)
 				lastOutSeq.append(IndexCommand(reachedIndex))
 			outSeqs.append([EndUtteranceCommand()])
-			return seq
 
+		paramsToReplay = []
```

Comment:
Might be good to move this to before the inner function. I'd like to get rid of that inner function, in fact `_processSpeechSequence` needs to be cleaned up.